rfc/rfc-78/rfc-78.md

```diff
@@ -0,0 +1,339 @@
+
+### CDC reads:
+- There has not been a lot of changes in 1.x wrt CDC. Only minor change was done in HoodieAppendHandle. but that affects only the writer side logic. So, we are good wrt CDC. i.e we do not need to port 
+any additional logic to 0.16.x reader just for the purpose of CDC in addition to changes covered in this RFC. 
+
+## 0.16.0 ➝ 1.0 upgrade steps
+This will be an automatic upgrade for users when they start using 1.x hudi library. Listing the changes we might need to do with the upgrade.
+- Rewrite archival timeline to LSM based timeline.
+- Do not touch active timeline, since there could be concurrent readers reading the table. So, 1.x reader should be capable of handling timeline w/ a mix of 0.x commit files and 1.x commit files.
+- But we need to trigger rollback of any failed writes using 0.16.x rollback logic. Since with 1.x, rollback will delete the log files based on delta commit times in log file naming, this rollback logic may not work for 
+log files written in 0.16.0, since we have log appends. So, we need to trigger rollbacks of failed writes using 0.16.x rollback flow and not 1.x flow. 
+- No changes to log reader.
+- Check custom payload class and hoodie.record.merge.mode in table properties and switch to respective 0.16.x properties.
+- Debatable: Trigger compaction for latest file slices. We do not want a single file slice having a mix of log files from 0.x and log files from 1.x. So, we will trigger a full compaction 
+of the table to ensure all latest file slices has just the base files. I need to dug deeper on this(need to have detailed discussion w/ initial authors who designed this). But if we ensure all partially failed writes are rollbacked completely, 
+we should be good to upgrade w/o needing to trigger full compaction. 
+
+Lets walk through an example:
+Say we have a file group and latest file slice is fs3. and it has 2 log files in 0.x. 
+  fs3(t3):
+     lf1_(3_10) lf2_(3_10)
+here format is lf[log version]_([committime]_[completiontime])
+base file instant time is 3 and so all log files (in memory) has same begin and completion time.(Remember completion time is only applicable for commits and not for data files). and so, from 1.x reader standpoint,
+lf1 and lf2 has delta commit time as t3 (which matches the base file's instant time) and its completion time is t10. 
+
+lets trigger an upgrade and add 2 new files. 
+
+fs3(t3):
+lf1_(3_10) lf2_(3_10) [upgrade] lf3_15_20, lf4_40_100.
+
+With this layout, 1.x reader should be able to read this file slice correctly w/o any issues. So, we should not require a full compaction during upgrade. But need to consult w/ authors to ensure we are not missing anything.
```

Comment:
I'd like to keep things simple. Its very likely that users are actively compacting  the table today anyway.